### PR TITLE
ARK-44 List Header and ARK-43 Author List

### PR DESCRIPTION
**Pull request by GeronimoJohn**
- Create list-header component
- Create components folder and move author-list into this folder
- Integrate list-header to author-list
- Refactor author-list props
     - Add AuthorLink, title, onSeeAllClick props

### CODE REVIEW

**Review by tataihono — CHANGES_REQUESTED** [review]
remove onClick from author list in favour of using AuthorLink prop. Also should use Link from material-ui/core. Happy to answer questions on how to do this.

File: src/compounds/author-list/author-list.composition.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+              avatar: {
+                url: 'https://www.jesusfilm.org/content/dam/jesusfilm/avatars/josh-mcdowell.jpg'
+              },
+              description:
+                'On staff with Cru as trailblazer for truth, Josh has been at the forefront of cultural trends and ground-breaking ministry for over five decades.'
+            }
+          }
+        }
+      ]}
+      title=""
```
Comment (tataihono) [inline src/compounds/author-list/author-list.composition.tsx:49]: this shouldn't be needed right?

File: src/compounds/author-list/author-list.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                avatar: {
+                  url: 'https://www.jesusfilm.org/content/dam/jesusfilm/avatars/josh-mcdowell.jpg'
+                },
+                description:
+                  'On staff with Cru as trailblazer for truth, Josh has been at the forefront of cultural trends and ground-breaking ministry for over five decades.'
+              }
+            }
+          }
+        ]}
+        title=""
```
Comment (tataihono) [inline src/compounds/author-list/author-list.spec.tsx:37]: should remove this

File: src/compounds/author-list/author-list.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      />
+    )
+    expect(getByText('Tez Brooks')).toBeInTheDocument()
```
Comment (tataihono) [inline src/compounds/author-list/author-list.spec.tsx:40]: this should also test that the header is not visible right?

File: src/elements/author-card/author-card.docs.mdx [hunk 4]
```diff
@@ -1,32 +1,29 @@
...
-  name="Tez Brooks"
-  description="Award winning author, screenwriter and international speaker"
-  src="https://www.jesusfilm.org/content/dam/jesusfilm/avatars/tezg.jpg"
-  onClick={() => alert("yoyoyo")}
+  author={{
+    node: {
+      name: 'Tez Brooks',
+      slug: 'tez-brooks',
+      avatar: {
+        url: ''
```
Comment (tataihono) [inline src/elements/author-card/author-card.docs.mdx:22]: can you put the image back for this one?

File: src/compounds/author-list/author-list.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    }
+  },
+  link: {
+    cursor: 'pointer'
+  }
+}))
+
+export type AuthorListProps = {
+  /** authors */
+  authors: AuthorCardProps[]
```
Comment (tataihono) [inline src/compounds/author-list/author-list.tsx:22]: this needs to just be the data for the author card.

```
  authors: {
    nodes: Author[]
  }
```

AuthorLink needs to be passed as a separate prop then passed down to the AuthorCard seperately. See the PremierePostListProps in Post List compound component for an example.

File: src/elements/author-card/author-card.composition.tsx [hunk 6]
```diff
@@ -3,18 +3,33 @@ import { AuthorCard, AuthorCardProps } from '.'
...
   <AuthorCard
-    name="Tez Brooks"
-    description="Award winning author, screenwriter and international speaker"
-    {...props}
+    author={{
+      node: {
+        name: 'Tez Brooks',
+        slug: 'tez-brooks',
+        avatar: {
+          url: ''
```
Comment (tataihono) [inline src/elements/author-card/author-card.composition.tsx:11]: you should just be able to drop the avatar attribute right?

File: src/elements/author-card/author-card.docs.mdx [hunk 7]
```diff
@@ -1,32 +1,29 @@
...
 description: "Present an author with an avatar and description."
 ---
 
 import { AuthorCard } from "./author-card";
 
-AuthorCard example:
+## React component that displays the author card
 
-Display a single author with name, description and profile image. 
+A component that display a single author with name, description and profile image.
```
Comment (tataihono) [inline src/elements/author-card/author-card.docs.mdx:10]:
```suggestion
A component that displays a single author with a name, description and profile image. 
```

File: src/elements/author-card/author-card.docs.mdx [hunk 8]
```diff
@@ -1,32 +1,29 @@
...
-<AuthorCard
-  name="Tez Brooks"
-  description="Award winning author, screenwriter and international speaker"
-  src="https://www.jesusfilm.org/content/dam/jesusfilm/avatars/tezg.jpg"
-  onClick={() => alert("yoyoyo")}
-/>
-```
-
-Live Playground:
+### Using props to customize the text
```
Comment (tataihono) [inline src/elements/author-card/author-card.docs.mdx:12]:
```suggestion
### Using props to customize the author
```

File: src/elements/list-header/list-header.docs.mdx [hunk 9]
```diff
@@ -0,0 +1,17 @@
+---
+description: 'A ListHeader component.'
+labels: ['Lists', 'React', 'Typescript']
+---
+
+import { ListHeader } from './list-header';
+
+## React Component our the list header
+
+A component that renders the header of a list it's under in
```
Comment (tataihono) [inline src/elements/list-header/list-header.docs.mdx:10]:
```suggestion
A component that renders a header for a list of items
```

File: src/elements/list-header/list-header.docs.mdx [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+### Using props to customize the text
```
Comment (tataihono) [inline src/elements/list-header/list-header.docs.mdx:12]:
```suggestion
### Using props to customize the title
```

File: src/elements/list-header/list-header.tsx [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+        <Grid item xs={9}>
+          <Typography className={classes.text} variant="h4">
+            {title}
+          </Typography>
+        </Grid>
+        <Grid item xs={3}>
+          {onSeeAllClick && (
+            <Box onClick={() => onSeeAllClick?.()}>
+              <Typography variant="body2" className={classes.seeAll}>
+                SEE ALL
```
Comment (tataihono) [inline src/elements/list-header/list-header.tsx:50]: translate this string

File: src/elements/list-header/list-header.tsx [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+  makeStyles
+} from '@material-ui/core'
+
+const useStyles = makeStyles((theme) => ({
+  divider: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(5)
+  },
+  text: {
+    color: '#EE3441',
```
Comment (tataihono) [inline src/elements/list-header/list-header.tsx:17]: I don't think we should set this here. Use one of the palette colors from the theme.

File: src/elements/list-header/list-header.tsx [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+      <Grid container spacing={4}>
+        <Grid item xs={9}>
+          <Typography className={classes.text} variant="h4">
+            {title}
+          </Typography>
+        </Grid>
+        <Grid item xs={3}>
+          {onSeeAllClick && (
+            <Box onClick={() => onSeeAllClick?.()}>
+              <Typography variant="body2" className={classes.seeAll}>
```
Comment (tataihono) [inline src/elements/list-header/list-header.tsx:49]: This should be a button with variant link.

File: src/elements/post/post.tsx [hunk 14]
```diff
@@ -1,30 +1,9 @@
 import React from 'react'
-import { AuthorCard } from '@jesus-film/ark.elements.author-card'
```
Comment (tataihono) [inline src/elements/post/post.tsx:2]: These changes are irrelevant to this PR.

**Review by tataihono — DISMISSED** [review]

File: src/elements/list-header/list-header.tsx (hunk 13 above)
Comment (GeronimoJohn) [reply src/elements/list-header/list-header.tsx:49]: When you say varian link. Do you mean like create a custom link like author link?

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**